lib/rides/pages/search_ride_page.dart

```diff
@@ -126,133 +103,296 @@ class _SearchRideFormState extends State<SearchRideForm> {
+              Image.asset('assets/pointing_up.png'),
+              const SizedBox(height: 10),
+              Text(
+                S.of(context).pageSearchRideNoInput,
+                style: Theme.of(context).textTheme.headlineMedium,
+                textAlign: TextAlign.center,
+              ),
+            ],
+          ),
+        );
+      }
+      return const SliverToBoxAdapter(
+        child: Padding(
+          padding: EdgeInsets.only(top: 10),
+          child: Center(
+            child: CircularProgressIndicator(),
+          ),
+        ),
+      );
+    }
+    final List<Ride> filteredSuggestions = _filter.apply(_rideSuggestions!, _selectedDate);
+    if (filteredSuggestions.isEmpty) {
+      return SliverToBoxAdapter(
+        child: SingleChildScrollView(
+          physics: const AlwaysScrollableScrollPhysics(),
+          child: Column(
+            children: <Widget>[
+              Image.asset('assets/shrug.png'),
+              const SizedBox(height: 10),
+              Text(
+                S.of(context).pageSearchRideEmpty,
+                style: Theme.of(context).textTheme.headlineMedium,
+                textAlign: TextAlign.center,
+              ),
+              if (_rideSuggestions!.isNotEmpty)
+                Semantics(
+                  button: true,
+                  tooltip: S.of(context).pageSearchRideTooltipFilter,
+                  child: InkWell(
+                    onTap: () => _filter.dialog(context, setState),
+                    child: Text(
+                      S.of(context).pageSearchRideRelaxRestrictions,
+                      style: Theme.of(context).textTheme.titleMedium,
+                      textAlign: TextAlign.center,
+                    ),
+                  ),
+                )
+              else
+                Text(
+                  S.of(context).pageSearchRideNoResults,
+                  style: Theme.of(context).textTheme.titleMedium,
+                  textAlign: TextAlign.center,
+                ),
+            ],
+          ),
+        ),
+      );
+    } else {
+      return SliverList(
+        delegate: SliverChildBuilderDelegate(
+          (BuildContext context, int index) {
+            final Ride ride = filteredSuggestions[index];
+            return RideCard(ride);
+          },
+          childCount: filteredSuggestions.length,
+        ),
+      );
+    }
+  }
+
+  Widget buildDateRow() {
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.spaceBetween,
+      children: <Widget>[
+        if (_wholeDay) ...<Widget>[
+          Row(
+            children: <Widget>[
+              IconButton(
+                onPressed: _selectedDate.isSameDayAs(DateTime.now())
+                    ? null
+                    : () => setState(() => _selectedDate = _selectedDate.subtract(const Duration(days: 1))),
+                icon: const Icon(Icons.chevron_left),
+              ),
+              Text(localeManager.formatDate(_selectedDate)),
```

Comment:
can we add that, we can select the datePicker here also? i think it's weird that that's different if i select whole day or not. we could leave the chevron icon just if i tap the text that the picker shows if i want to look more than a few days in the future thats way easier with the picker.

Reply:
Yes actually that's one thing we also already learned from the user studies. That will definitely be added.